client/scss/components/_button.scss

```diff
@@ -57,32 +57,32 @@
   }
 
   &.warning {
```

Comment:
It seems we don't have this in our Styleguide. Would be nice to add, though not necessarily have to be in this PR.

We'll need to add a new button and a new dropdown button, e.g.

<img width="487" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/bda81a11-862b-4092-a456-725bbb36237c">

<img width="404" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/e668a6f5-95e7-4fcb-be32-a3930f12466b">

(May need their own subsections inside the button/dropdown button sections.)

In practice, we can see the warning button (or at least the warning dropdown buttons) when reverting a revision.